### ISSUE TITLE
Model Based RL

### ISSUE THREAD

**Issue by seancraven** [seancraven/mono_ms#7]
I need to get first some kind of model-based RL going to learn transition dynamics for an MDP homomorphism:

The rough idea here is to learn transition and reward functions that behave like Group Structured MDP homomorphisms then perform normal rl in $\overline{\mathcal{M}}$ and lift the policy down!

**Comment by seancraven** [comment]
![plots](https://github.com/seancraven/mono_ms/assets/85280921/1e04ef9f-07c9-4ac1-a366-be16673c714d)

**Comment by seancraven** [comment]
The above plot is a PPO Actor-Critic agent trained on a Neural network that predicts the next state from a state action pair. 

- The rules for cart pole termination and reward are hardcoded and not learnt
- Clearly, there is a large performance delta between the "Planned Reward" and the received reward from the actual mdp

**Comment by seancraven** [comment]
![plots](https://github.com/seancraven/mono_ms/assets/85280921/ec25db1b-1364-4db2-8686-04798d3ac07a)

**Comment by seancraven** [comment]
For reference these are model free curves

**Comment by seancraven** [comment]
I have a feeling that I might have to use something like Dyna or another hybrid method where the transition model learns online with the actor critic.

**Comment by seancraven** [comment]
What do I currently know about why this is not working well?

**Comment by seancraven** [comment]
Plan to approach:
- Investigating fit quality: look at the loss curves per dimension. 
- Investigating dataset quality, are the traditions I have sampled just unrepresentative?

**Comment by seancraven** [comment]
Plot of model training in an offline manner
